### PR TITLE
Ignoring ! char in ACL name of use_backend clause

### PR DESCRIPTION
**Pull request by flyhard**
Should be fixing #2 temporarily until a better solution is found

Also added ability to generate a Hash for each object for comparisson and extended parsing a bit

### CODE REVIEW

File: acl.go [hunk 1]
```diff
@@ -48,6 +49,10 @@ func backendReferenceByAcl(frontend Frontend, backends []Backend) error {
 		for _, a := range ub.Condition.Any {
 			// TODO support or/and conditions
 			for _, s := range a {
+				//TODO Handle ! correctly instead of ignoring it
+				if strings.HasPrefix(s, "!") {
+					s=strings.TrimLeft(s,"!")
```
Comment (takumakanari) [inline acl.go]: Could you apply `gofmt` to follow Go's coding style?

File: hashFuncs.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package haproxyconfigparser
```
Comment (takumakanari) [inline hashFuncs.go]: I think rename `hashFuncs.go` to `hashfuncs.go` is more natural.

**Review by takumakanari — COMMENTED** [review]
Hi @flyhard.
Firstly, thanks for your contribution.

I've just added 2 comments. Can you check it?
Thanks.

File: acl.go (hunk 1 above)
Comment (flyhard) [reply acl.go]: Sure. Thanks for the suggestion

File: hashFuncs.go (hunk 2 above)
Comment (flyhard) [reply hashFuncs.go]: Sure will.

File: acl.go (hunk 1 above)
Comment (takumakanari) [reply acl.go]: 👍

**Comment by takumakanari** [comment]
Thanks for your adding great work!
Can I merge it now?

**Comment by flyhard** [comment]
Yes, go ahead

**Comment by takumakanari** [comment]
Merged.
Thank you.

**Pull request merged**